webapp/src/pages/ResponsePage/ResponsePage.js

```diff
@@ -0,0 +1,209 @@
+            if (withdrawn) {
+                return ["withdrawn" + " " + data.withdrawn_timestamp]
+            }
+        }
+    }
+
+
+
+    // Render Sections
+    renderSections() {
+        const applicationForm = this.state.applicationForm;
+        const applicationData = this.state.applicationData;
+        let html = [];
+        // Translation
+        const t = this.props.t;
+
+        // main function
+        if (applicationForm && applicationData) {
+            applicationForm.sections.forEach(section => {
+                html.push(<div key={section.name} className="section">
+                    { /*Heading*/}
+                    <div className="flex baseline"><h3>{section.name}</h3></div>
+                    { /*Q & A*/}
+                    <div className="Q-A">
+                        {this.renderQuestions(section)}
+                    </div>
+                </div>)
+            })
+        }
+
+        return html
+    }
+
+
+
+    // Render Questions 
+    renderQuestions(section) {
+        let questions = section.questions.map(q => {
+            return <div key={q.id} className="question-answer-block">
+                <p>{q.headline}</p>
+                <h6>{this.renderAnswer(q.id, q.type)}</h6>
+            </div>
+        })
+        return questions
+    }
+
+
+
+    // Render Answers 
+    renderAnswer(id, type) {
+        const applicationData = this.state.applicationData;
+        const baseUrl = process.env.REACT_APP_API_URL;
+        let answers;
+
+        applicationData.answers.forEach(a => {
+            if (a.question_id == id) {
+                formatAnswer(a, type)
+            }
+        })
+
+        // format aswerers 
+        function formatAnswer(a, type) {
+            // file
+            if (type == "file") {
+                answers = <a className="answer file" key={a.value} target="_blank" href={baseUrl + "/api/v1/file?filename=" + a.value}>{a.value}</a>
+            }
+            // multi-file
+            if (type == "multi-file") {
+                let files = [];
+                a.value.forEach((file => {
+                    if (file) {
+                        files.push(
+                            <div key={a.headline}><a key={a.headline} target="_blank" href={baseUrl + "/api/v1/file?filename=" + file} className="answer">{a.value}</a></div>
+                        )
+                    }
+                }))
+                answers = <div key={a.headline}>{files}</div>
+            }
+            // choice
+            if (type.includes("choice")) {
+                let choices = [];
+                a.options.forEach(opt => {
+                    if (a.value == opt.valuel) {
+                        choices.push(<div key={opt.label}><label className="answer">{opt.label}</label></div>)
+                    }
+                })
+                answers = <div key={choices}>{choices}</div>
+            }
+            // text
+            if (type.includes("text")) {
```

Comment:
Can you make this the default, because there are a few other types not catered for in your current list which we don't want to be ignored - so I suggest making these ifs into "if, else if, else"

Reply:
I think I follow your logic, I have rearranged the statements to "text" is the first if and all other queries below are else if statement, then the last statement I added is this `   else {
                        handleAnswers.push([{
                            headline: answer.headline, value: <div key={answer.headline}><p>{answer.value}</p>
                            </div>
                        }])
                    }`